pkg/controller/job/job_controller_test.go

```diff
@@ -58,11 +58,19 @@ import (
 
 var alwaysReady = func() bool { return true }
 
+func newJobWithName(name string, parallelism, completions, backoffLimit int32, completionMode batch.CompletionMode) *batch.Job {
+	return generateJob(name, parallelism, completions, backoffLimit, completionMode)
+}
+
 func newJob(parallelism, completions, backoffLimit int32, completionMode batch.CompletionMode) *batch.Job {
+	return generateJob("foobar", parallelism, completions, backoffLimit, completionMode)
+}
+
+func generateJob(name string, parallelism, completions, backoffLimit int32, completionMode batch.CompletionMode) *batch.Job {
```

Comment:
nit: just have two functions `newJob` and `newJobWithName`. `newJob` calls `newJobWithName`.